app/controllers/api/v1/publications_controller.rb

```diff
@@ -226,6 +236,92 @@ def grants
           key :api_key, []
         end
       end
+
+      operation :patch do
+        key :summary, 'Update Publication'
+        key :description, 'Update publication\'s ScholarSphere Open Access Link'
+        key :operationId, 'updatePublication'
+        key :produces, [
+          'application/json'
+        ]
+        key :tags, [
+          'publication'
+        ]
+        parameter do
+          key :name, :Publication
+          key :in, :body
+          key :description, 'Publication to update with the provided ScholarSphere Open Access Link'
+          key :required, true
+          schema do
+            key :'$ref', :PublicationInput
+          end
+        end
+        response 200 do
+          key :description, 'ScholarSphere Open Access Link successfully updated response'
+          schema do
+            key :'$ref', :UpdateOpenAccessLocationsResult
+          end
+        end
+        response 404 do
+          key :description, 'No publications found response'
+          schema do
+            key :'$ref', :ErrorModelV1
+          end
+        end
+        response 422 do
+          key :description, 'ScholarSphere Open Access Link already exists response'
+          schema do
+            key :'$ref', :ErrorModelV1
+          end
+        end
+        response 401 do
+          key :description, 'unauthorized'
+          schema do
+            key :'$ref', :ErrorModelV1
+          end
+        end
+        security do
+          key :api_key, []
+        end
+      end
+    end
+
+    swagger_schema :PublicationInput do
+      one_of do
```

Comment:
@EricDurante I use `one_of` here to define the schema thinking that the input would reflect as either containing a doi or activity_insight_id but did not work as the way I expected. Any thoughts on how to define the schema here?

Reply:
@banukutlu I think I remember running into this situation before when I was writing docs for SchlarSphere's API. I gave up trying to figure how how to make it "one of these" and just put in a descriptive note. I think if the error message is clear, "You provided both A and B, but you can only use A or B" or something along those lines.

Reply:
I am kinda at that point, I tried couple variations and also oneOf is not supported by 2.0. 

So, should we just leave the doc as below in that case? I am not sure how to represent it 🤔 

```
{
 "doi": "string"
  "activity_insight_id": "string",
  "scholarsphere_open_access_url": "string"
}
```

Reply:
@banukutlu 
Yeah, I totally agree with Adam. I would maybe just note this restriction in the description of the publication parameter and then also make sure that the error message is descriptive like Adam suggested. I think that will be enough to clarify the proper usage.